src/H5T.c

```diff
@@ -5288,14 +5261,6 @@ H5T__path_find_real(const H5T_t *src, const H5T_t *dst, const char *name, H5T_co
         table          = path;
     } /* end else-if */
 
-    /* Set the flag to indicate both source and destination types are compound types
-     * for the optimization of data reading (in H5Dio.c).
-     * Make sure that path->are_compounds is only true for compound types.
-     */
-    path->are_compounds = false;
```

Comment:
This field wasn't used anywhere and is trivial to add back in the future if needed